test/mlir/krnl/krnl_category_mapper.mlir

```diff
@@ -0,0 +1,193 @@
+// CHECK-DAG:   llvm.mlir.global internal constant @G(dense<[1, 0, -3]> : vector<3xi32>) : !llvm.array<3 x i32>
+
+// CHECK-LABEL: llvm.func @test_find_index_int(%arg0: i64) -> i64
+// CHECK-DAG:   [[LEN:%.+]] = llvm.mlir.constant(3 : i32) : i32
+// CHECK-DAG:   [[G:%.+]] = llvm.extractvalue {{.*}}[1] : !llvm.struct<(ptr<i32>, ptr<i32>, i64, array<1 x i64>, array<1 x i64>)>
+// CHECK-DAG:   [[V:%.+]] = llvm.extractvalue {{.*}}[1] : !llvm.struct<(ptr<i32>, ptr<i32>, i64, array<1 x i64>, array<1 x i64>)>
+// CHECK:       [[INDEX:%.+]] = llvm.call @find_index_i64(%arg0, [[G]], [[V]], [[LEN]]) : (i64, !llvm.ptr<i32>, !llvm.ptr<i32>, i32) -> i32
+// CHECK:       [[UNR_CONV:%.+]] = builtin.unrealized_conversion_cast [[INDEX]] : i32 to i64
+// CHECK:       llvm.return [[UNR_CONV]] : i64
+}
+
+// -----
+
+// Test CategorMapper lowering when the input is a list of strings.
+func private @test_category_mapper_string_to_int64(%arg0: memref<2x2x!krnl.string>) -> memref<2x2xi64> {
+  %c0_i32 = arith.constant 0 : i32
+  %c-1_i64 = arith.constant -1 : i64
+  %c3_i32 = arith.constant 3 : i32
+  %0 = memref.alloc() {alignment = 16 : i64} : memref<2x2xi64>
+  %1 = "krnl.global"() {name = "G0", shape = [3], value = dense<[1, 0, -3]> : vector<3xi32>} : () -> memref<3xi32>
+  %2 = "krnl.global"() {name = "V1", shape = [3], value = dense<[1, 2, 0]> : vector<3xi32>} : () -> memref<3xi32>
+  %3 = "krnl.global"() {name = "cats_int64s2", shape = [3], value = dense<[1, 2, 3]> : tensor<3xi64>} : () -> memref<3xi64>
+  %4 = "krnl.global"() {name = "cats_strings3", shape = [3], value = dense<["cat", "dog", "cow"]> : tensor<3x!krnl.string>} : () -> memref<3x!krnl.string>
+  %5:2 = krnl.define_loops 2
+  krnl.iterate(%5#0, %5#1) with (%5#0 -> %arg1 = 0 to 2, %5#1 -> %arg2 = 0 to 2) {
+    %6:2 = krnl.get_induction_var_value(%5#0, %5#1) : (!krnl.loop, !krnl.loop) -> (index, index)
+    %7 = krnl.load %arg0[%6#0, %6#1] : memref<2x2x!krnl.string>
+    %8 = "krnl.find_index"(%7, %1, %2, %c3_i32) : (!krnl.string, memref<3xi32>, memref<3xi32>, i32) -> index
+    %9 = krnl.load %4[%8] : memref<3x!krnl.string>
+    %10 = "krnl.strlen"(%9) : (!krnl.string) -> i64
+    %11 = "krnl.strncmp"(%7, %9, %10) : (!krnl.string, !krnl.string, i64) -> i32
+    %12 = arith.cmpi eq, %11, %c0_i32 : i32
+    scf.if %12 {
+      %13 = krnl.load %3[%8] : memref<3xi64>
+      krnl.store %13, %0[%6#0, %6#1] : memref<2x2xi64>
+    } else {
+      krnl.store %c-1_i64, %0[%6#0, %6#1] : memref<2x2xi64>
+    }
+  }
+  return %0 : memref<2x2xi64>
+
+  // CHECK-DAG: llvm.func @strncmp(!llvm.ptr<i8>, !llvm.ptr<i8>, i64) -> i32
+  // CHECK-DAG: llvm.func @strlen(!llvm.ptr<i8>) -> i64
+  // CHECK-DAG: llvm.func @find_index_str(!llvm.ptr<i8>, !llvm.ptr<i32>, !llvm.ptr<i32>, i32) -> i32
+  // CHECK-DAG: llvm.mlir.global internal constant {{.*}}(dense<["cat", "dog", "cow"]> : tensor<3x!krnl.string>) : !llvm.array<3 x struct<(ptr<i8>, ptr<i8>, i64, array<1 x i64>, array<1 x i64>)>>
```

Comment:
Just want to make sure that `!krnl.string` type can exit from MLIR and be lowered to LLVM properly. Have you checked that this generated code can be translated to LLVM and then to Bitcode?